### PR TITLE
Ensures nested properties with dump_only are excluded

### LINKED ISSUE #82: dump_only is ignored in nested schemas for parameters
Given the following:

``` python
class ParentSchema(Schema):
    id = fields.Int(dump_only=True)
    name = fields.String()
    child = fields.Nested('ChildSchema')

class ChildSchema(Schema):
    id = fields.Int(dump_only=True)
    name = fields.String()
```

The parameter spec is generated:  

``` json
"parameters": [
  {
    "in": "body", 
    "name": "body", 
    "required": false, 
    "schema": {
      "properties": {
        "child": {
          "properties": {
            "id": {
              "format": "int32", 
              "type": "integer"
            }, 
            "name": {
              "type": "string"
            }
          }, 
          "type": "object"
        }, 
        "name": {
          "type": "string"
        }
      }, 
      "type": "object"
    }
  }
]
```

The Parent "id" is omitted but the Child "id" remains. 

Will submit a PR on this.

### PR DESCRIPTION
**Pull request by incognick**
Resolves #82

### CODE REVIEW

File: apispec/ext/marshmallow/swagger.py [hunk 1]
```diff
@@ -250,7 +250,7 @@ def field2property(field, spec=None, use_refs=True, dump=True, name=None):
         elif spec:
             ret.update(resolve_schema_dict(spec, field.schema))
```
Comment (incognick) [inline apispec/ext/marshmallow/swagger.py:251]: line 251:  Does this line need `dump=dump` as well?  My test passes without so I didn't add this.

**Comment by sloria** [comment]
Thanks!

File: apispec/ext/marshmallow/swagger.py (hunk 1 above)
Comment (sloria) [inline apispec/ext/marshmallow/swagger.py:251]: Yes, turns out it should be passed here as well; done in https://github.com/marshmallow-code/apispec/commit/cbe3f33b1f89dab2124d41d663e7633c8d6c8617

**Comment by incognick** [comment]
Awesome! Thank you

**Pull request merged**